### ISSUE TITLE
Which normalization transform whould I use to use pre-trained models?

### ISSUE THREAD

**Issue by GeorgeBatch** [binli123/dsmil-wsi#42]
Hi,

I want to use your pre-trained ResNet models, but I could not find which transforms you used. To be more precise, when using ImageNet pre-trained models on new inputs, people normalize the input images before passing them to the pre-trained extractor using ImageNet normalization constants (mean=[0.485, 0.456, 0.406] and std=[0.229, 0.224, 0.225]).

What normalization should I perform to use your pre-trained models?

Many thanks,
George Batchkala

**Comment by binli123** [comment]
I tried using ImageNet pretrained ResNet18 on the TCGA dataset without normalizing the input image patches, it worked decently well. Make sure the model uses BatchNorm (not InstanceNorm which is the one used for SimCLR training) and the model is in evaluation mode when computing the features.

**Comment by GeorgeBatch** [comment]
Thanks! For people who will read this issue later, to follow @binli123's advice, add **--norm_layer 'batch'** when running `python compute_feats.py`. By default, `--norm_layer` is 'instance'.

**Comment by xiachenrui** [comment]
But I notice in your paper, the pre-trained ResNet18 have a bad performance in Table3. Can you explain this more detailly?

Thanks !

<img width="357" alt="image" src="https://github.com/binli123/dsmil-wsi/assets/47804819/ca8c6d81-ab8d-4447-af37-f7d7f6756ac4">

> I tried using ImageNet pretrained ResNet18 on the TCGA dataset without normalizing the input image patches, it worked decently well. Make sure the model uses BatchNorm (not InstanceNorm which is the one I used in SimCLR training) and the model is in evaluation mode when computing the features.

**Comment by ZechengLi19** [comment]
> I tried using ImageNet pretrained ResNet18 on the TCGA dataset without normalizing the input image patches, it worked decently well. Make sure the model uses BatchNorm (not InstanceNorm which is the one I used in SimCLR training) and the model is in evaluation mode when computing the features.

I'm curious why you use InstanceNorm instead of BatchNorm, can you share your thoughts?
